docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
+
+Requirements
+------------
+
+- Collect both containers and system logs (STDOUT or STDERR)
+- Persistent storage of logs
+- Horizontally scalable (capacity and performance)
+- Accessible from a central end point
+- Highly Available
+- Easy to use query language
+- Service accessible through a UI
+- Monitorable service (health, performances and alerts)
+- Ability to correlate logs with KPIs and alerts
+- Multitenancy for audit logs
+- Performances (TBD)
+
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  system error investigation
+
+- As a Platform Administrator, I want to browse all system logs (from all
+  servers) from a unique endpoint, in order to ease distributed system error
+  investigation
+
+- As a Storage Administrator, I want to browse all Hyperdrive containers logs
+  (from all servers) from a unique endpoint, in order to ease Hyperdrive error
+  investigation
+
+- As a Storage Administrator, I want to browse all Zenko containers logs (from
+  all servers) from a unique endpoint, in order to ease Zenko error
+  investigation
+
+- As a Storage Administrator, I want to log all actions performed by users
+  (whatever their Role) on the RING X product, in order to be able to sustain
+  an audit
+
+
+(some other US extracted from Loki design doc)
+
+- After receiving an alert on my service and drilling into the query associated
+  with said alert, I want to quickly see the logs associated with the jobs
+  which produced those timeseries at the time of the alert.
+
+- After a pod or node disappears, I want to be able to retrieve logs from just
+  before it died, so I can diagnose why it died.
+
+- After discovering an ongoing issue with my service, I want to extract a
+  metric from some logs and combine it with my existing time series data.
+
+- I have a legacy job which does not expose metrics about errors - it only logs
+  them.  I want to build an alert based on the rate of occurrences of errors in
+  the log.
+
+
+Documentation
+-------------
+
+Deployment & Configuration
+--------------------------
+
+indexing log streams instead of indexing log content
```

Comment:
?

Reply:
to be removed ... it is a draft :) but this is the key difference between Loki and Kibana/ES. I wonder if it is going to be a problem or not for workload logs browsing

Reply:
That's an implementation detail, so not part of the requirements IMO. If we want full-text indexing on the logs, let's state it in the requirements and discard Loki today. If we don't want such full-text search, let's state it as well to make sure it is approved before we start the design.

Reply:
updated